core/src/starknet/filter.rs

```diff
@@ -45,13 +45,91 @@ impl Filter {
     }
 
     /// Add message to filter.
-    pub fn add_message<F>(mut self, closure: F) -> Self
+    pub fn add_message<F>(&mut self, closure: F) -> &mut Self
     where
         F: Fn(L2ToL1MessageFilter) -> L2ToL1MessageFilter,
     {
         self.messages.push(closure(L2ToL1MessageFilter::default()));
         self
     }
+
+    /// Build final version of Filter
+    pub fn build(&mut self) -> Self {
+        self.clone()
+    }
+}
+
+impl TransactionFilter {
+    /// Create `InvokeTransactionV0Filter` from `TransactionFilter`
+    pub fn invoke_transaction_v0<F>(&mut self, closure: F) -> &mut Self
```

Comment:
What if these methods simply returned the new builder? I believe that would improve the API quite a bit.

Reply:
as discussed together, this does not seems to be easily feasible for the moment